aiobotocore/httpxsession.py

```diff
@@ -0,0 +1,305 @@
+            # since arbitrary dots and slashes are valid as key paths.
+            # See test_basic_s3.test_non_normalized_key_paths
+            # This way of using it is currently ~undocumented, but recommended in
+            # https://github.com/encode/httpx/discussions/1805#discussioncomment-8975989
+            extensions = {"target": bytes(url, encoding='utf-8')}
+
+            httpx_request = self._session.build_request(
+                method=request.method,
+                url=url,
+                headers=headers,
+                content=content,
+                extensions=extensions,
+            )
+            # auth, follow_redirects
+            response = await self._session.send(httpx_request, stream=True)
+            response_headers = botocore.compat.HTTPHeaders.from_pairs(
+                response.headers.items()
+            )
+
+            http_response = aiobotocore.awsrequest.HttpxAWSResponse(
+                str(response.url),
+                response.status_code,
+                response_headers,
+                response,
+            )
+
+            if not request.stream_output:
+                # Cause the raw stream to be exhausted immediately. We do it
+                # this way instead of using preload_content because
+                # preload_content will never buffer chunked responses
+                await http_response.content
+
+            return http_response
+
+        # **previous exception mapping**
+        # aiohttp.ClientSSLError -> SSLError
+
+        # aiohttp.ClientProxyConnectiorError
+        # aiohttp.ClientHttpProxyError -> ProxyConnectionError
+
+        # aiohttp.ServerDisconnectedError
+        # aiohttp.ClientPayloadError
+        # aiohttp.http_exceptions.BadStatusLine -> ConnectionClosedError
+
+        # aiohttp.ServerTimeoutError -> ConnectTimeoutError|ReadTimeoutError
+
+        # aiohttp.ClientConnectorError
+        # aiohttp.ClientConnectionError
+        # socket.gaierror -> EndpointConnectionError
+
+        # asyncio.TimeoutError -> ReadTimeoutError
+
+        # **possible httpx exception mapping**
+        # httpx.CookieConflict
+        # httpx.HTTPError
+        # * httpx.HTTPStatusError
+        # * httpx.RequestError
+        #   * httpx.DecodingError
+        #   * httpx.TooManyRedirects
+        # * httpx.TransportError
+        #   * httpx.NetworkError
+        #     * httpx.CloseError -> ConnectionClosedError
+        #     * httpx.ConnectError -> EndpointConnectionError
+        #     * httpx.ReadError
+        #     * httpx.WriteError
+        #   * httpx.ProtocolError
+        #     * httpx.LocalProtocolError -> SSLError??
+        #     * httpx.RemoteProtocolError
+        #   * httpx.ProxyError -> ProxyConnectionError
+        #   * httpx.TimeoutException
+        #     * httpx.ConnectTimeout -> ConnectTimeoutError
+        #     * httpx.PoolTimeout
+        #     * httpx.ReadTimeout -> ReadTimeoutError
+        #     * httpx.WriteTimeout
```

Comment:
given all timeouts previously went to ConnectTimeoutError probably we can just map `TimeoutException` to that

Reply:
can use your judgement looking at the botocore implementation